src/Domain/Criteria/Criteria.php

```diff
@@ -6,37 +6,81 @@ class Criteria
 {
     private $filters;
     private $order;
+    private $sorting;
     private $offset;
     private $limit;
 
-    public function __construct(Filters $filters, ?Order $order, ?int $offset, ?int $limit)
+    public function __construct(Filters $filters, $sorting, ?int $offset, ?int $limit)
```

Comment:
Podriamos usar Sorting|Order|null ?
```suggestion
    public function __construct(Filters $filters, Sorting|Order|null $sorting, ?int $offset, ?int $limit)
```

Reply:
no se tipa de momento para soportar php 7.4, se podrá meter después de eliminar el soporte de 7.4